aws/resource_aws_kinesis_analyticsv2_application.go

```diff
@@ -0,0 +1,2177 @@
+		kinesisanalyticsv2.MetricsLevelApplication,
+		kinesisanalyticsv2.MetricsLevelOperator,
+		kinesisanalyticsv2.MetricsLevelParallelism,
+		kinesisanalyticsv2.MetricsLevelTask,
+	}, false)
+
+	validateKinesisAnalyticsV2LogLevel = validation.StringInSlice([]string{
+		kinesisanalyticsv2.LogLevelInfo,
+		kinesisanalyticsv2.LogLevelWarn,
+		kinesisanalyticsv2.LogLevelError,
+		kinesisanalyticsv2.LogLevelDebug,
+	}, false)
+
+	validateKinesisAnalyticsV2ConfigurationType = validation.StringInSlice([]string{
+		kinesisanalyticsv2.ConfigurationTypeCustom,
+		kinesisanalyticsv2.ConfigurationTypeDefault,
+	}, false)
+)
+
+func resourceAwsKinesisAnalyticsV2Application() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsKinesisAnalyticsV2ApplicationCreate,
+		Read:   resourceAwsKinesisAnalyticsV2ApplicationRead,
+		Update: resourceAwsKinesisAnalyticsV2ApplicationUpdate,
+		Delete: resourceAwsKinesisAnalyticsV2ApplicationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				arns := strings.Split(d.Id(), ":")
+				name := strings.Replace(arns[len(arns)-1], "application/", "", 1)
+				d.Set("name", name)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"service_execution_role": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"runtime": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateKinesisAnalyticsV2Runtime,
+			},
+
+			"s3_bucket": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"s3_object": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"code": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"code_content_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"create_timestamp": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"last_update_timestamp": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"version": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"cloudwatch_logging_options": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"log_stream_arn": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateArn,
+						},
+					},
+				},
+			},
+
+			// Flink only
+			"property_groups": {
```

Comment:
Given that there can be more than one of these and order doesn't seem to be important, `Type: schema.TypeSet` seems more appropriate.